docs/manual/upgrades.md

```diff
@@ -0,0 +1,129 @@
+The `UpdatePodConfig` subreconciler will ensure that all Pods have the new configuration present, this configuration ensures that the `fdbmonitor` will restart processes with the new binary present in the shared volume.
+Once all Pods have the new configuration and binaries present the operator can move to the next phase.
+
+#### Restart Phase
+
+The `BounceProcesses` subreconciler will handle the restart of all `fdbserver` processes.
+For FoundationDB clusters that are spanned across multiple Kubernetes clusters the operator will follow a special process which is document in the [technical design](technical_design.md#bounceprocesses).
+The subreconciler will also ensure to wait until all `fdbserver` processes are ready to be restarted to prevent cases where only a subset of processes are restarted.
+After the `kill` command the operator will initiate a new reconciliation loop to detect the new running version, this is handled in the `UpdateState` subreconciler and the version is detected based on the output of the [cluster status json](https://apple.github.io/foundationdb/mr-status.html).
+
+#### Recreation of Pods Phase
+
+--> Add info incompatbile clients
+
+
+Technically the upgrade is now done and the cluster is running on the new version but the operator has to do some cleanup.
+The cleanup includes the recreation of all Pods to ensure they use the new FoundationDB image for the desired version and the configuration will be updated to point again to the binary present in the container.
+Depending on the defined [Pod update strategy](../cluster_spec.md#podupdatestrategy) the operator will handle the recreation of the Pods differently.
+In the default setting `ReplaceTransactionSystem` the operator will recreate the storage Pods zone by zone and replaces Pods of the transaction system (log and stateless processes).
+The replacement of the transaction system has the benefit that the recoveries of the transaction system is reduced to potentially one recovery.
+Once all Pods are updated to the new image the upgrade is done and the cluster status of the FoundationDB cluster resource in Kubernetes should show that the reconciliation is done.
+
+### Known issues
+
+There are a number of known issues that can occur during an upgrade of FoundationDB running on Kubernetes.
+Most of the issues are only a challenged for version incompatible upgrades and most issues are handled automatically by the operator.
+If an issue cannot be handled by the operator we will outline a way to manually resolve the issue.
+
+#### Process restarted during staging Phase
+
+During the staging phase it can happen that a `fdbserver` process gets restarted by an external event like the OOM killer or any issue that results in the `fdbserver` crashing.
+In that case the `fdbmonitor` will restart the `fdbserver` process with the new version and for version incompatible upgrades, that means this process is not able to connect to the cluster.
```

Comment:
If that process happens to be a stateless process then this will cause a recovery, correct? I wonder how recovery will work if the restarted process is at a later version? Especially, if this process happens to be a tlog process (as the sequencer has to read state from tlogs in order to proceed with recovery)? And, what if multiple tlogs (equal to the tlog replication factor) restart, can the sequencer even complete recovery?

Reply:
Correct, if the process is a log process this will issue also a recovery. 

> I wonder how recovery will work if the restarted process is at a later version? Especially, if this process happens to be a tlog process (as the sequencer has to read state from tlogs in order to proceed with recovery)? And, what if multiple tlogs (equal to the tlog replication factor) restart, can the sequencer even complete recovery?

Those are good questions I cannot answer directly and we have to dig deeper into the recovery process. In our test case where we keep half of the stateless and log processes at the old version, we see that the cluster will come up and recover.